zipkin-autoconfigure/storage-mysql/src/main/java/zipkin2/autoconfigure/storage/mysql/ZipkinMySQLStorageProperties.java

```diff
@@ -87,18 +97,27 @@ public void setUseSsl(boolean useSsl) {
   }
 
   public DataSource toDataSource() {
-    StringBuilder url = new StringBuilder("jdbc:mysql://");
-    url.append(getHost()).append(":").append(getPort());
-    url.append("/").append(getDb());
-    url.append("?autoReconnect=true");
-    url.append("&useSSL=").append(isUseSsl());
-    url.append("&useUnicode=yes&characterEncoding=UTF-8");
     HikariDataSource result = new HikariDataSource();
     result.setDriverClassName("org.mariadb.jdbc.Driver");
-    result.setJdbcUrl(url.toString());
+    result.setJdbcUrl(determineJdbcUrl());
     result.setMaximumPoolSize(getMaxActive());
     result.setUsername(getUsername());
     result.setPassword(getPassword());
     return result;
   }
+
+  private String determineJdbcUrl() {
+    StringBuilder url = new StringBuilder();
+    if (StringUtils.hasText(getJdbcUrl())) {
+      url.append(getJdbcUrl());
```

Comment:
nit: there's no value to using stringbuilder here. just exit early then retain existing logic if not present

Reply:
👍